payments/tests/test_bambora_payform_refund.py

```diff
@@ -69,6 +69,58 @@ def test_initiate_refund_success(provider_base_config: dict, order: Order):
     )
 
 
+@pytest.mark.parametrize(
+    "order", ["berth_order", "winter_storage_order"], indirect=True,
+)
+def test_initiate_refund_no_order_email(provider_base_config: dict, order: Order):
+    """Test the request creator constructs the payload base and returns a url that contains a token"""
+    request = RequestFactory().request()
+    order.status = OrderStatus.PAID
+    order.email = None
+    order.lease.status = LeaseStatus.PAID
+    order.lease.save()
+    order.save()
+
+    OrderToken.objects.create(
```

Comment:
This token is no longer valid? It doesn't have effect on test result though

Reply:
It doesn't. We just use the token to ensure that it was generated through Bambora. Since the payment could've been issued long time ago, it doesn't really matter if it's valid or not, as long as it exists.

Reply:
Actually, if there are valid tokens, that means that the payment has not been made/settled, since we expire all the tokens when we receive the confirmation from VismaPay. It's not very likely to happen anyway, but I just noticed that.

Reply:
Ok, solved, I pushed a commit that also covers this case.